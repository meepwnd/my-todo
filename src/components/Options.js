import React from "react";
import Option from "./Option";
import { connect } from "react-redux";

export const Options = props => (
  <ul className="options">
    {props.options.map(option => (
      <Option
        key={option.id}
        option={option.option}
        removeOption={props.removeOption}
        toggleCompletion={props.toggleCompletion}
        id={option.id}
        completed={option.completed}
      />
    ))}
  </ul>
);

const filterOptions = (options, filter) => {
  switch (filter) {
    case "SHOW_COMPLETED":
      return options.filter(option => option.completed === true);
    case "SHOW_UNCOMPLETED":
      return options.filter(option => option.completed === false);
    case "SHOW_ALL":
      return options;
    default:
      return options;
  }
};
const mapStateToProps = state => ({
  options: filterOptions(state.options, state.filter)
});

export default connect(mapStateToProps)(Options);
