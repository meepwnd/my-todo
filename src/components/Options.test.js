import React from "react";
import { shallow } from "enzyme";
import { Options } from "./Options";

describe("options component", () => {
  it("should render correctly with two options", () => {
    const props = {
      options: [
        {
          key: 1,
          option: "option 1",
          removeOption: () => {},
          toggleCompletion: () => {},
          id: "1",
          completed: true
        },
        {
          key: 2,
          option: "option 2",
          removeOption: () => {},
          toggleCompletion: () => {},
          id: "2",
          completed: true
        }
      ]
    };
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly with no options", () => {
    const props = {
      options: []
    };
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
