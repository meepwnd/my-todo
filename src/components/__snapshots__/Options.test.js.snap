// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options component should render correctly with no options 1`] = `
<ul
  className="options"
/>
`;

exports[`options component should render correctly with two options 1`] = `
<ul
  className="options"
>
  <Connect(Option)
    completed={true}
    id="1"
    key="1"
    option="option 1"
  />
  <Connect(Option)
    completed={true}
    id="2"
    key="2"
    option="option 2"
  />
</ul>
`;
